### ISSUE TITLE
Buggy behavior with empty dict literal and string:

### ISSUE THREAD

**Issue by squeaky-pl** [zheller/flake8-quotes#97]
Consider the following example:

```
def test():
    {}["a"]
```

and this configuration:

```
[flake8]
max-line-length = 119
select = Q0
inline-quotes = double
```

this ends up with `Q002 Remove bad quotes from docstring` when there is no docstring here involved.

Splitting this over two lines like:

```
def test():
    d = {}
    d["a"]
```

Resolves it.

**Comment by twolfson** [comment]
Oh wow, interesting edge case. I'll look into reproducing this either today
or tomorrow

**Comment by twolfson** [comment]
I've successfully reproduced the issue, at first glance it looks like an issue in our custom docstring detection function but I'll need to dig deeper which I don't have time for at the moment. I'll take a shot at fixing this by Sunday

Also going to @ @oxitnik since they've worked on this code last/might not have forgotten as much as I have

**Comment by oxitnik** [comment]
Yes. issue in get_docstring_tokens. `state` needs to be reset when brackets are handled.
I tried to replace https://github.com/zheller/flake8-quotes/blob/26fe39c9b9a160a4dc2f3c975f71f0479bc193b7/flake8_quotes/docstring_detection.py#L64-L67 with
```python
        elif token.type == tokenize.OP and token.string in ['(', '[', '{']:
            bracket_count += 1
            if state in [STATE_EXPECT_MODULE_DOCSTRING, STATE_EXPECT_CLASS_DOCSTRING,
                       STATE_EXPECT_FUNCTION_DOCSTRING]:
                state = STATE_OTHER
        elif token.type == tokenize.OP and token.string in [')', ']', '}']:
            bracket_count -= 1
            if state in [STATE_EXPECT_MODULE_DOCSTRING, STATE_EXPECT_CLASS_DOCSTRING,
                       STATE_EXPECT_FUNCTION_DOCSTRING]:
                state = STATE_OTHER
```
and no docstrings have been found in snippet from issue.

**Comment by twolfson** [comment]
:+1: That patch works great =D Will do my best to properly attribute the commit in a PR

**Comment by twolfson** [comment]
This has been released in `3.2.0`. Thanks for the bug report and fixes y'all =D
